### PR TITLE
feat: add running user to all-contributors

### LINKED ISSUE #156: 🚀 Feature: Setup script should add the running user to .all-contributorsrc
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/template-typescript-node-package/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

From #147: if someone is running the setup script, they should be added to the `.all-contributorsrc` [allcontributors config](https://allcontributors.org/docs/en/bot/configuration) for (I'm thinking to start): `code`, `content`, `documentation`, `ideas`, `infrastructure`, `maintenance`, `projectManagement`, `tool`.

### Additional Info

See https://allcontributors.org/docs/en/emoji-key for the full allcontributors emoji key.

The setup script already makes sure the user is authenticated with [`gh`](https://cli.github.com). So you can get information on the user by running `gh api user` in the script.

```ts
const user = JSON.parse(await $`gh api user`);
```

### PR DESCRIPTION
**Pull request by RNR1**
## PR Checklist

- [x] Addresses an existing open issue: fixes #156
- [x] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/template-typescript-node-package/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/template-typescript-node-package/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

I added some logic to add the running user to the `all-contributors` list (according to the authenticated GitHub CLI account).
At first, I tried only to use the owner variable that the user prompted, but it failed the tests, where a fabricated owner name was provided; if you prefer to only use the provided owner name without reaching out to gh CLI, we might need to adjust the tests a little bit.

### CODE REVIEW

File: script/setup.js [hunk 1]
```diff
@@ -169,6 +169,9 @@ try {
 
 	await withSpinner(
 		async () => {
+			const user = JSON.parse((await $`gh api user`).stdout);
+			const userName = user.login;
+			await $`all-contributors add ${userName} code,content,doc,ideas,infra,maintenance,projectManagement,tool`;
```
Comment (JoshuaKGoldberg) [inline script/setup.js:174]: > prefer to only use the provided owner name without reaching out to gh CLI

```plaintext
    stderr: 'gh: To use GitHub CLI in a GitHub Actions workflow, set the GH_TOKEN environment variable. Example:\n' +
```

Hmm, yeah, this is a situation... here are some starting thoughts:
* If `gh api user` _is_ available, we'd want to use it
* Running `all-contributors add` is nice for validating that the user does in fact exist
* If `gh api user` _isn't_ available, it makes sense to me to fall back on the provided owner name

What do you think about having the setup script trying, in order...
1. `gh api user` & `all-contributors add`
2. Provided owner & `all-contributors add` - and logging a warning
3. Throwing an error

...where we hardcode a username that we can expect to exist as the provided owner? Maybe ... yours? 😄

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
A good start!

File: script/setup.js (hunk 1 above)
Comment (RNR1) [reply script/setup.js]: Thanks for the feedback, I applied the following changes, let me know what you think

File: script/setup.js [hunk 2]
```diff
@@ -178,14 +206,13 @@ try {
...
-						contributors: [
-							{
-								...existingContributors.contributors.find(
-									({ login }) => login === "JoshuaKGoldberg"
-								),
-								contributions: ["tool"],
-							},
-						],
+						contributors: existingContributors.contributors
+							.filter(({ login }) => [user, "JoshuaKGoldberg"].includes(login))
```
Comment (JoshuaKGoldberg) [inline script/setup.js:210]: [Praise] Very clean and succinct, I like it 😄

File: script/setup.js [hunk 3]
```diff
@@ -167,8 +167,36 @@ try {
...
+
 	await withSpinner(
 		async () => {
+			let user;
+			try {
+				user = JSON.parse((await $`gh api user`).stdout).login;
+			} catch (err) {
+				console.warn(
+					chalk.gray(
+						"Couldn't authenticate GitHub user, falling back to the owner name you provided"
```
Comment (JoshuaKGoldberg) [inline script/setup.js:192]: [Nit] Just to make it a teeny bit more clear... what do you think?

```suggestion
						"Couldn't authenticate GitHub user, falling back to the provided owner name '${owner}'"
```

File: script/setup.js [hunk 4]
```diff
@@ -167,8 +167,36 @@ try {
...
+		"doc",
+		"ideas",
+		"infra",
+		"maintenance",
+		"projectManagement",
+		"tool",
+	];
+
+	const addContributor = (user, contributions = []) =>
+		$`all-contributors add ${user} ${contributions.join(",")}`;
```
Comment (JoshuaKGoldberg) [inline script/setup.js:182]: [Refactor] Why separate these out into an earlier place in code if `addContributor` is only ever called once? Why not have the <code>await $\`...\`</code> inline in the one usage place?

```js
await $`all-contributors add ${user} ${[ /* ... */ ].join(",")}`;
```

**Review by JoshuaKGoldberg — APPROVED** [review]
Looks great to me, thanks! Just a single stylistic question - otherwise no complaints. Will wait for your answer before merging. ✨

**Comment by RNR1** [comment]
Thanks, I applied these changes

**Pull request merged**